ui/src/parseError.ts

```diff
@@ -68,13 +69,19 @@ export function parseError(error: any): IParsedError {
 
     message = parseIfHtml(message);
 
+    let actionContext: Partial<IActionContext> | undefined;
+    if (error instanceof OverwriteActionContextError) {
```

Comment:
This is one reason I want this as an interface - you should simply check for the existence of the actionContext property, rather than checking for a specific derived class.  That's what duck typing is for.

Reply:
Are you suggesting something like below? if so, how would type script ensures that the error.actionContext is of type IActionContext?
`    if (error.actionContext) {
        actionContext = error.actionContext;
    }`

Reply:
If it has the properties of a type, it *is* of that type.

I'd suggest:
const errorActionContext = (<{ actionContext?: Partial<IActionContext> }>)error.actionContext;
if (typeof errorActionContext === "object") { ... }

Now, this doesn't ensure that the properties under errorActionContext are as expected, although at that point. For instance, if error.actionContext.telemetry were a string, then referencing errorActionContext.telemetry.properties would throw an error.  That'd be pretty unlikely, you can probably just ignore it, or wrap that code in a try/catch. Or you could be explicit with type checking everything as I did above but personally I'd go with try/catch for everything under actionContext.

Reply:
Not Applicable with the new design